### ISSUE TITLE
[Enhancement Request]: allow method prefixes for `ICommand`

### ISSUE THREAD

**Issue by powerdude** [CommunityToolkit/dotnet#162]
I would previously name methods i was using for a command with either `On<command>` or `On<command>Async`.  Like `ObservableProperty` that allows `_` or `m_` prefixes, I think it would be nice to allow method prefixes.

**Comment by Sergio0694** [comment]
This kinda makes sense to me, but also not sure if it's a good idea. As in, why the need to use the "On" prefix at all? 🤔
Then again I'm not personally familiar with how common this pattern is.

cc. @michael-hawker @Arlodotexe thoughts?

**Comment by egvijayanand** [comment]
This makes complete sense to me. Have been following this pattern of On prefixed method names for quite a while as it's easy to distinguish the logic methods and command methods.

**Comment by michael-hawker** [comment]
@Sergio0694 it's getting a bit hard to follow all the bits and bobs that are going on with the source generators. Is there a list/doc somewhere you're keeping up-to-date?

Currently we take

```cs
[ICommand]
private void SomeCommandName() {}
```

and make a `SomeCommandNameCommand` property that's a RelayCommand that can be used, right?

And if you name it `SomeCommandNameAsync` it still makes it `SomeCommandNameCommand` right?

Therefore, this proposal is that if you do the following:

```cs
[ICommand]
private void OnSomeCommandName() { }
```

Then `SomeCommandNameCommand` would still be the generated name?

Did I get that right?

(What happens if your method ends in `Command` do we duplicate it or ignore it?)

As I could see something where you name your method like `OnFileSaveCommand` and want a `FileSaveCommand` property?

@powerdude @egvijayanand do you have some real-world examples of things you've used in the past like this to help fill in he details?

**Comment by powerdude** [comment]
I think of these methods like event handlers that you would usually find in the view code-behind and named as `ButtonName_Clicked`.  Since we're in the VM, we don't usually add the `_Clicked` to method names since we try to keep UI artifacts out of the VM.  After using Prism and ReactiveUI, I've seen the pattern of prefixing event handler-like methods with `On` so that they sort of "stick out" and when code is formatted by tools like Resharper, they would end up grouped together as private/protected event handlers.  For me, it is a great way to help segment the behaviors in a view model.  

I would never name the method `OnButtonNameCommand` though as I don't think `Command` belongs in the method/event handler name.

**Comment by Sergio0694** [comment]
@michael-hawker 

> "Therefore, this proposal is that if you do the following:
>
>```csharp
>[ICommand]
>private void OnSomeCommandName() { }
>```
>Then SomeCommandNameCommand would still be the generated name?"

Correct, the proposal is that the "On" prefix would just be stripped from the generated commands.
Essentially just the same as how "Async" suffixes is also skipped, just with the "On" prefix instead 🙂

**Comment by egvijayanand** [comment]
Exactly. As of now, the `Async` suffix is removed while generating the command name.

Similarly, the `On` prefix should be removed from the method name marked with the `[ICommand]` attribute while generating the command representing it.

```cs
[ICommand]
private void OnSubmit()
{
}

public IRelayCommand SubmitCommand { get; }

[ICommand]
private async Task OnLoginAsync()
{
}

public IAsyncRelayCommand LoginCommand { get; }
```
